### ISSUE TITLE
Deprecations are not visible in clients

### ISSUE THREAD

**Issue by spaenleh** [graasp/graasp-query-client#926]
## Issue

When marking a hook as deprecated in query-client it does not show in the client code. 
This is due to two things:
- we do not keep the comments 
- we spread the return from the hooks, so we loose the deprecation

## Solution
If we can remove the need for the `configureHooks` this issue can be resolved.
For this we need to remove the need for the `API_HOST` env variable.
This can be done by calling the api on the same host, by removing the crossOrigin issue.
If we remove the crossOrigin calls we can also remove the usage of the `withCredentials` that is added in the `configureAxios` function. 

In summary if we move to a single origin architecture with the api running on the `/api` subpath and apps running on their own paths (`/builder` etc).

To solve the issue of the local development we could use [the proxy feature from the vite development server](https://vitejs.dev/config/server-options#server-proxy) which would allow to have requests to `/api` go to `http://localhost:3000` instead of `http://localhost:3111/api` (for builder for example).
